sway-lib-std/src/bytes.sw

```diff
@@ -0,0 +1,849 @@
+    pub fn new() -> Self {
+        Self {
+            ptr: alloc_bytes(0),
+            cap: 0,
+        }
+    }
+
+    /// Creates a `RawBytes` (on the heap) with exactly the capacity (in bytes) specified.
+    /// This is equivalent to calling `RawBytes::new` when `capacity` is `0`.
+    pub fn with_capacity(capacity: u64) -> Self {
+        Self {
+            ptr: alloc_bytes(capacity),
+            cap: capacity,
+        }
+    }
+
+    /// Gets the pointer of the allocation.
+    pub fn ptr(self) -> raw_ptr {
+        self.ptr
+    }
+
+    /// Gets the capacity of the allocation.
+    pub fn capacity(self) -> u64 {
+        self.cap
+    }
+
+    /// Grow the capacity of the Bytes by doubling its current capacity. The
+    /// `realloc_bytes` function allocates memory on the heap and copies
+    /// the data from the old allocation to the new allocation.
+    pub fn grow(ref mut self) {
+        let new_cap = if self.cap == 0 { 1 } else { 2 * self.cap };
+        self.ptr = realloc_bytes(self.ptr, self.cap, new_cap);
+        self.cap = new_cap;
+    }
+}
+
+pub struct Bytes {
+    buf: RawBytes,
+    len: u64,
+}
+
+impl Bytes {
+    /// Constructs a new, empty `Bytes`.
+    ///
+    /// The Bytes will not allocate until elements are pushed onto it.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std::bytes::Bytes;
+    ///
+    /// let bytes = Bytes::new();
+    /// // does not allocate
+    /// assert(bytes.len() == 0);
+    /// assert(bytes.capacity() == 0);
+    /// ```
+    pub fn new() -> Self {
+        Bytes {
+            buf: RawBytes::new(),
+            len: 0,
+        }
+    }
+
+    /// Constructs a new, empty `Bytes` with the specified capacity.
+    ///
+    /// The Bytes will be able to hold exactly `capacity` bytes without
+    /// reallocating. If `capacity` is 0, the Bytes will not allocate.
+    ///
+    /// It is important to note that although the returned Bytes has the
+    /// capacity specified, the vector will have a zero length.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std::bytes::Bytes;
+    ///
+    /// let bytes = Bytes::with_capacity(2);
+    /// // does not allocate
+    /// bytes.push(5);
+    /// // does not re-allocate
+    /// bytes.push(10);
+    /// ```
+    pub fn with_capacity(capacity: u64) -> Self {
+        Bytes {
+            buf: RawBytes::with_capacity(capacity),
+            len: 0,
+        }
+    }
+
+    /// Appends an element to the back of a Bytes collection.
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// use std::bytes::Bytes;
+    ///
+    /// let mut bytes = Bytes::new();
+    /// let a = 5u8;
+    /// let b = 7u8;
+    /// bytes.push(a);
+    /// bytes.push(b);
+    /// assert(bytes.len() == 2);
+    /// ```
+    pub fn push(ref mut self, byte: u8) {
+        // If there is insufficient capacity, grow the buffer.
+        if self.len == self.buf.capacity() {
+            self.buf.grow();
+        };
```

Comment:
```suggestion
        }
```

Reply:
I've been working under the assumption that a semicolon after an `if` statement is needed _unless_ I want to implicitly return the resulting value?

Reply:
This is no longer a problem and these semicolons are only required if you have something like `let x = if cond { 5 } else { 6 };`